### ISSUE TITLE
String parse error when executing project run with override arguments

### ISSUE THREAD

**Issue by paulrinckens** [explosion/spaCy#7947]
## How to reproduce the behaviour

When running a spacy project with override arguments, e.g.

````
python -m spacy project clone pipelines/tagger_parser_ud
python -m spacy project run all --vars.train_name test
````

this will throw the following error:

````
    commands = {cmd["name"]: cmd for cmd in config.get("commands", [])}
TypeError: string indices must be integers
````

Looks to me as if the override value is injected into the config as '"test"' instead of 'test' which leads to failed parse of the config back from string due to broken command section: https://github.com/explosion/spaCy/blob/49aed683cce4d58baca10e7cb4fe89fbfc209a36/spacy/cli/_util.py#L194

## Info about spaCy

- **spaCy version:** 3.0.5
- **Platform:** macOS-11.2.3-x86_64-i386-64bit
- **Python version:** 3.9.2
- **Pipelines:** en_core_web_md (3.0.0), en_core_web_sm (3.0.0)

**Comment by svlandeg** [comment]
Hi!

Thanks for the report. I think that this bug should be resolved in a very recent [PR](https://github.com/explosion/thinc/pull/488), which was released in Thinc 8.0.3. Could you try updating `thinc` and test your command again?

For me locally, with the latest versions, this command now works:
```
python -m spacy project run all . --vars.train_name test
```

**Comment by paulrinckens** [comment]
@svlandeg thank you for the quick check! Yes, the issue seems to be resolved with thinc 8.0.3.
